### ISSUE TITLE
Hosting and Continuous Integration/Deployment (CI/CD)

### ISSUE THREAD

**Issue by DotNetDublin** [BlazingSaddlesSolutions/TimeKeeper#12]
We need to examine hosting options for our application and associated databases.

Depending on the outcome of issue #1 we could be looking at hosting for the following

- Blazor WASM Application
- or
- Blazor Application which combines WASM and Server Side
- ASP.Net Core API
- SQL Server Database

The above is obviously subject to change.

As a starting point I am suggesting utilising Azure and implementing continuous integration.

In order to mimic a real world scenario it would be great to have a Staging and a Production environment.

I would imagine due to account restrictions not everyone will be able to access the Azure account however those that do will document the step taken to achieve continuous integration and everyone will be able to see that when changes are pushed to set branches in the repository these are automatically fed through to the application.

**Comment by jonhilt** [comment]
Great summary of where we're at so far :-)

Azure seems like a good starting point when considering hosting, not
least because it's a platform many of us are likely to find ourselves
using at some point if we're developing .NET applications.

On May 6, 2021, GitHub ***@***.***> wrote:
> We need to examine hosting options for our application and associated
> databases.
>
> Depending on the outcome of issue #1
> <https://github.com/BlazingSaddlesSolutions/TimeKeeper/issues/1> we
> could be looking at hosting for the following
>
>  *  Blazor WASM Application
>  *  or
>  *  Blazor Application which combines WASM and Server Side
>  *  ASP.Net Core API
>  *  SQL Server Database
>
> The above is obviously subject to change.
>
> As a starting point I am suggesting utilising Azure and implementing
> continuous integration.
>
> In order to mimic a real world scenario it would be great to have a
> Staging and a Production environment.
>
> I would imagine due to account restrictions not everyone will be able
> to access the Azure account however those that do will document the
> step taken to achieve continuous integration and everyone will be able
> to see that when changes are pushed to set branches in the repository
> these are automatically fed through to the application.
>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/BlazingSaddlesSolutions/TimeKeeper/issues/12>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAAZDA7QUJF62TQWM5YF673TMLR6ZANCNFSM44HV3ZNA>.

**Comment by harperjohn** [comment]
I agree! At first I "resisted" Azure... I felt like I was being "corralled" (and possibly "lead to slaughter").

But then I gave it a try... so far I have been successful at staying within the "free" limits... Not the I am unwilling to pay for hosting, it was I was afraid that all the little "micro-costs" would snowball and get out of control.

And given this time to play with the free level... I really do like the close integration with Blazor/Visual Studio!

I have just used the "publish to Azure" option and it worked very nicely. I have not yet tried a more comprehensive CI/CD yet, but I can see that it is the way to go.
